web/src/stream_data.js

```diff
@@ -351,24 +351,27 @@ export function muted_stream_ids() {
         .map((sub) => sub.stream_id);
 }
 
-export function get_subscribed_streams_for_user(user_id) {
+export function get_sub_or_unsub_streams_for_user(user_id, get_unsubscribed = false) {
```

Comment:
Let's call this `get_streams_for_user(user_ids)` and have it return an object with two keys `subscribed` and `can_subscribe`.